lucene/core/src/java/org/apache/lucene/codecs/lucene80/Lucene80DocValuesProducer.java

```diff
@@ -0,0 +1,1430 @@
+          return disi.nextDoc();
+        }
+
+        @Override
+        public int docID() {
+          return disi.docID();
+        }
+
+        @Override
+        public long cost() {
+          return disi.cost();
+        }
+
+        @Override
+        public int advance(int target) throws IOException {
+          set = false;
+          return disi.advance(target);
+        }
+
+        @Override
+        public boolean advanceExact(int target) throws IOException {
+          set = false;
+          return disi.advanceExact(target);
+        }
+
+        @Override
+        public long nextOrd() throws IOException {
+          if (set == false) {
+            final int index = disi.index();
+            final long start = addresses.get(index);
+            this.start = start + 1;
+            end = addresses.get(index + 1L);
+            set = true;
+            return ords.get(start);
+          } else if (start == end) {
+            return NO_MORE_ORDS;
+          } else {
+            return ords.get(start++);
+          }
+        }
+
+      };
+    }
+  }
+
+  @Override
+  public void checkIntegrity() throws IOException {
+    CodecUtil.checksumEntireFile(data);
+  }
+
+  /**
+   * Reader for longs split into blocks of different bits per values.
+   * The longs are requested by index and must be accessed in monotonically increasing order.
+   */
+  // Note: The order requirement could be removed as the jump-tables allow for backwards iteration
+  // Note 2: The rankSlice is only used if an advance of > 1 block is called. Its construction could be lazy
+  private class VaryingBPVReader {
+    final RandomAccessInput slice; // 2 slices to avoid cache thrashing when using rank
+    final RandomAccessInput rankSlice;
+    final NumericEntry entry;
+    final int shift;
+    final long mul;
+    final int mask;
+
+    long block = -1;
+    long delta;
+    long offset;
+    long blockEndOffset;
+    LongValues values;
+
+    VaryingBPVReader(NumericEntry entry) throws IOException {
+      this(entry,
+          data.randomAccessSlice(entry.valuesOffset, entry.valuesLength),
+          data.randomAccessSlice(entry.valuesOffset, entry.valuesLength));
+    }
+    VaryingBPVReader(NumericEntry entry, RandomAccessInput slice, RandomAccessInput rankSlice) throws IOException {
+      this.entry = entry;
+      this.slice = slice;
+      this.rankSlice = rankSlice;
+      shift = entry.blockShift;
+      mul = entry.gcd;
+      mask = (1 << shift) - 1;
+    }
+
+    long getLongValue(long index) throws IOException {
+      final long block = index >>> shift;
+      if (this.block != block) {
+        int bitsPerValue;
+        do {
+          // If the needed block is the one directly following the current block, it is cheaper to avoid the cache
+          if (entry.valueJumpTableOffset != -1 && block != this.block+1) {
+            blockEndOffset = rankSlice.readLong(entry.valueJumpTableOffset +block*Long.BYTES-entry.valuesOffset)-
```

Comment:
can we start the slice at `entry.valueJumpTableOffset` to avoid adding it every time (related to the above comment)